iothub/service/src/DirectMethod/DirectMethodRequest.cs

```diff
@@ -41,34 +41,27 @@ public class DirectMethodRequest
         /// in this request having a timeout of 0 seconds.
         /// </remarks>
         [JsonIgnore]
-        public TimeSpan ConnectionTimeout { get; set; }
+        public TimeSpan? ConnectionTimeout { get; set; }
 
         [JsonProperty("responseTimeoutInSeconds", NullValueHandling = NullValueHandling.Ignore)]
-        internal int? ResponseTimeoutInSeconds => ResponseTimeout <= TimeSpan.Zero
-            ? null
-            : ResponseTimeout.Seconds;
+        internal int? ResponseTimeoutInSeconds => ResponseTimeout?.Seconds;
```

Comment:
I think you want TotalSeconds

Reply:
<https://www.tutorialspoint.com/difference-between-timespan-seconds-and-totalseconds>

Reply:
Nice catch. That's not very intuitive design by Timespan :/